fuel/datasets/hdf5.py

```diff
@@ -64,3 +67,127 @@ def get_data(self, state=None, request=None):
                 raise ValueError
         data = [node[request] for node in self.nodes]
         return data
+
+
+@do_not_pickle_attributes('data_sources')
+class H5PYDataset(Dataset):
+    """An h5py-fueled HDF5 dataset.
+
+    This dataset class assumes a particular file layout:
+
+    * Data sources reside in the root group, and their names define the
+      source names.
+    * The dataset is not explicitly split. Instead, splits are defined as
```

Comment:
Alright, well, I don't totally understand the corpora argument and I can't think of when you'd ever want to do this for a dataset like MNIST. If we go this direction, we should test this _very_ thoroughly. Nothing more frustrating then realizing you've accidentally trained on the test data and your results are bullshit. :)